**Python for Public Policy Final Project**

Let’s explore the 911-end-to-end calls in New York City
Download data from https://data.cityofnewyork.us/Public-Safety/911-End-to-End-Data/t7p9-n9dy/data

**1.1 Research Question**


From 2016 to 2019, which agency responded to the lowest number of incidents?

**1.2 Hypothesis**


NYPD, because FDNY is known to receive a high volume of calls on a daily basis.

**1.2.1 Import the necessary package**


For the purpose of this assignment, we will be using Pandas

In [1]:
import pandas as pd


**1.2.2 Read and save 911-End-to-End dataset as a pandas dataframe**


Let’s name this dataframe “emergencies”

In [2]:
emergencies= pd.read_csv('911_End-to-End_Data.csv')

**1.2.3 Today’s Goal**

Learn which incidents were the least common and discover which agencies responded to the least number of incidents from 2016 to 2019. But first, let’s take a look at the data and clean it up.

**1.2.4 Preview the data contents**

In [3]:
 emergencies.head() #This shows the first 5 rows in the dataset as a default

Week Start Date Agency                      Final Incident Type  \
0      12/29/2014    EMS        1. Life Threating Med Emergencies   
1      12/29/2014    EMS  2. Non-Life Threatening Med Emergencies   
2      12/29/2014   FDNY                      1. Structural Fires   
3      12/29/2014   FDNY                  2. Non-Structural Fires   
4      12/29/2014   FDNY                   3. Medical Emergencies   

   # of Incidents Calculated  Call to First Pickup  \
0                       8881                  3.02   
1                      12594                  3.03   
2                        341                  3.09   
3                        168                  3.07   
4                       3054                  3.02   

   Call to PD Calltaker Handoff  Call to FDNY Pickup  \
0                         60.84                  NaN   
1                         69.63                  NaN   
2                         71.82                  NaN   
3                         80.86                  NaN   
4                        154.83                  NaN   

   Call to FDNY Job Creation  Call to EMS Pickup  Call to Agency Job Creation  \
0                        NaN               68.10                       128.32   
1                        NaN               77.76                       125.53   
2                        NaN                 NaN                          NaN   
3                        NaN                 NaN                          NaN   
4                        NaN               66.45                       151.56   

   ...  Median Travel  \
0  ...          329.5   
1  ...          441.0   
2  ...          190.0   
3  ...          235.0   
4  ...          243.0   

   Median Cumulative First Arrival  (Multi-Agency Incidents)  Average Travel  \
0                                             427.75                    6.05   
1                                             571.50                    8.42   
2                                             267.00                    3.25   
3                                             327.00                    4.20   
4                                             375.00                    4.26   

   Average Dispatch  Average EMS Processing  Average EMS Pickup  \
0              1.58                    0.87                0.15   
1              2.67                    0.66                0.16   
2              0.20                     NaN                 NaN   
3              0.24                     NaN                 NaN   
4              0.39                    1.45                0.10   

   Average Calltaker Processing  Average Pickup  Average FD Pickup  \
0                          0.97            0.05                NaN   
1                          1.11            0.05                NaN   
2                          1.15            0.05                NaN   
3                          1.30            0.05                NaN   
4                          2.53            0.05                NaN   

   Average FD Processing  
0                    NaN  
1                    NaN  
2                    NaN  
3                    NaN  
4                    NaN  

[5 rows x 30 columns]

In [4]:
emergencies.tail() #This shows the last 5 rows in the dataset as a default

Week Start Date          Agency                      Final Incident Type  \
6343      01/07/2019  NYPD (Non-CIP)               Other Crimes (In Progress)   
6344      01/07/2019  NYPD (Non-CIP)                               Past Crime   
6345      01/07/2019  NYPD (Non-CIP)  Police Officer/Security Holding Suspect   
6346      01/07/2019  NYPD (Non-CIP)                          Possible Crimes   
6347      01/07/2019  NYPD (Non-CIP)                         Vehicle Accident   

      # of Incidents Calculated  Call to First Pickup  \
6343                       2720                  3.58   
6344                       3200                  3.72   
6345                        195                  3.57   
6346                       7361                  3.56   
6347                       3179                  3.69   

      Call to PD Calltaker Handoff  Call to FDNY Pickup  \
6343                        152.76                  NaN   
6344                        205.11                  NaN   
6345                         99.61                  NaN   
6346                        148.94                  NaN   
6347                        173.26                  NaN   

      Call to FDNY Job Creation  Call to EMS Pickup  \
6343                        NaN                 NaN   
6344                        NaN                 NaN   
6345                        NaN                 NaN   
6346                        NaN                 NaN   
6347                        NaN                 NaN   

      Call to Agency Job Creation  ...  Median Travel  \
6343                          NaN  ...          305.0   
6344                          NaN  ...         1531.0   
6345                          NaN  ...          658.0   
6346                          NaN  ...         1031.0   
6347                          NaN  ...         1730.0   

      Median Cumulative First Arrival  (Multi-Agency Incidents)  \
6343                                              599.0           
6344                                             2236.0           
6345                                              977.0           
6346                                             1549.0           
6347                                             2388.0           

      Average Travel  Average Dispatch  Average EMS Processing  \
6343            9.52              3.13                     NaN   
6344           44.40              9.24                     NaN   
6345           18.62              3.71                     NaN   
6346           33.97              7.78                     NaN   
6347           44.66              9.59                     NaN   

      Average EMS Pickup  Average Calltaker Processing  Average Pickup  \
6343                 NaN                          2.49            0.06   
6344                 NaN                          3.36            0.06   
6345                 NaN                          1.60            0.06   
6346                 NaN                          2.42            0.06   
6347                 NaN                          2.83            0.06   

      Average FD Pickup  Average FD Processing  
6343                NaN                    NaN  
6344                NaN                    NaN  
6345                NaN                    NaN  
6346                NaN                    NaN  
6347                NaN                    NaN  

[5 rows x 30 columns]

In [5]:
emergencies.size #This shows how many cells are in the data table

190440

In [6]:
emergencies.info() #This shows information about each column

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6348 entries, 0 to 6347
Data columns (total 30 columns):
 #   Column                                                     Non-Null Count  Dtype  
---  ------                                                     --------------  -----  
 0   Week Start Date                                            6348 non-null   object 
 1   Agency                                                     6348 non-null   object 
 2   Final Incident Type                                        6348 non-null   object 
 3   # of Incidents Calculated                                  6348 non-null   int64  
 4   Call to First Pickup                                       6348 non-null   float64
 5   Call to PD Calltaker Handoff                               6348 non-null   float64
 6   Call to FDNY Pickup                                        384 non-null    float64
 7   Call to FDNY Job Creation                                  384 non-null    float64
 8   Call to 

Let’s look at which agencies are included in this dataset

In [7]:
emergencies['Agency'].unique()

array(['EMS', 'FDNY', 'NYPD', 'NYPD (Non-CIP)'], dtype=object)

Let’s look at the list of final incident types

In [8]:
incidents=emergencies['Final Incident Type'].unique()

Let’s make it easier to read

In [9]:
incidents.sort()
list(incidents)

['1. Critical',
 '1. Life Threating Med Emergencies',
 '1. Structural Fires',
 '2. Non-Life Threatening Med Emergencies',
 '2. Non-Structural Fires',
 '2. Serious',
 '3. Medical Emergencies',
 '3. Non-Critical',
 '4. Non-Medical Emergencies',
 'Alarms',
 'Disorderly Person/Group/Noise',
 'Dispute',
 'Hazardous Materials/Suspicious Letters/Packages/Substances/Substances',
 'Investigate/Possible Crime',
 'Other Crimes (In Progress)',
 'Past Crime',
 'Police Officer/Security Holding Suspect',
 'Possible Crimes',
 'Shot Spotter',
 'United States Postal Service - Bio Hazard Detection System',
 'Vehicle Accident']

Let’s group the records in the dataset based on their Final Incident Type value

In [10]:
emergencies.groupby('Final Incident Type').size()

Final Incident Type
1. Critical                                                              349
1. Life Threating Med Emergencies                                        351
1. Structural Fires                                                      349
2. Non-Life Threatening Med Emergencies                                  351
2. Non-Structural Fires                                                  349
2. Serious                                                               349
3. Medical Emergencies                                                   349
3. Non-Critical                                                          349
4. Non-Medical Emergencies                                               349
Alarms                                                                   349
Disorderly Person/Group/Noise                                            349
Dispute                                                                  349
Hazardous Materials/Suspicious Letters/Packages/Substanc

**1.2.5 Data Cleaning**

Since we want to take a look at data only from 2016 to 2019, we need to create a dataframe with data from only this period. First, let’s check the mininum and maximum week start dates.

In [11]:
emergencies['Week Start Date'].min()

'01/01/2018'

In [12]:
emergencies['Week Start Date'].max()

'12/31/2018'

There is a problem here. The above functions seem to not work. It is capturing the lowest and highest value, but it is not identifying the line as a date. Let’s try converting columns to datetime timestamps.

In [13]:
emergencies['Week Start Date']=pd.to_datetime(emergencies['Week Start Date'],format= '%m/%d/%Y')

In [14]:
emergencies.head()

Week Start Date Agency                      Final Incident Type  \
0      2014-12-29    EMS        1. Life Threating Med Emergencies   
1      2014-12-29    EMS  2. Non-Life Threatening Med Emergencies   
2      2014-12-29   FDNY                      1. Structural Fires   
3      2014-12-29   FDNY                  2. Non-Structural Fires   
4      2014-12-29   FDNY                   3. Medical Emergencies   

   # of Incidents Calculated  Call to First Pickup  \
0                       8881                  3.02   
1                      12594                  3.03   
2                        341                  3.09   
3                        168                  3.07   
4                       3054                  3.02   

   Call to PD Calltaker Handoff  Call to FDNY Pickup  \
0                         60.84                  NaN   
1                         69.63                  NaN   
2                         71.82                  NaN   
3                         80.86                  NaN   
4                        154.83                  NaN   

   Call to FDNY Job Creation  Call to EMS Pickup  Call to Agency Job Creation  \
0                        NaN               68.10                       128.32   
1                        NaN               77.76                       125.53   
2                        NaN                 NaN                          NaN   
3                        NaN                 NaN                          NaN   
4                        NaN               66.45                       151.56   

   ...  Median Travel  \
0  ...          329.5   
1  ...          441.0   
2  ...          190.0   
3  ...          235.0   
4  ...          243.0   

   Median Cumulative First Arrival  (Multi-Agency Incidents)  Average Travel  \
0                                             427.75                    6.05   
1                                             571.50                    8.42   
2                                             267.00                    3.25   
3                                             327.00                    4.20   
4                                             375.00                    4.26   

   Average Dispatch  Average EMS Processing  Average EMS Pickup  \
0              1.58                    0.87                0.15   
1              2.67                    0.66                0.16   
2              0.20                     NaN                 NaN   
3              0.24                     NaN                 NaN   
4              0.39                    1.45                0.10   

   Average Calltaker Processing  Average Pickup  Average FD Pickup  \
0                          0.97            0.05                NaN   
1                          1.11            0.05                NaN   
2                          1.15            0.05                NaN   
3                          1.30            0.05                NaN   
4                          2.53            0.05                NaN   

   Average FD Processing  
0                    NaN  
1                    NaN  
2                    NaN  
3                    NaN  
4                    NaN  

[5 rows x 30 columns]

Check data types and confirm they are now datetime

In [15]:
emergencies.dtypes

Week Start Date                                              datetime64[ns]
Agency                                                               object
Final Incident Type                                                  object
# of Incidents Calculated                                             int64
Call to First Pickup                                                float64
Call to PD Calltaker Handoff                                        float64
Call to FDNY Pickup                                                 float64
Call to FDNY Job Creation                                           float64
Call to EMS Pickup                                                  float64
Call to Agency Job Creation                                         float64
Call to Agency Dispatch                                             float64
Call to Agency Arrival                                              float64
Call to First Arrival (Multi-Agency Incidents)                      float64
Median Picku

In [16]:
emergencies['Week Start Date'].min()

Timestamp('2013-11-18 00:00:00')

In [17]:
emergencies['Week Start Date'].max()

Timestamp('2020-08-24 00:00:00')

This worked, but the timestamp is now there. Let’s pull out just the date and create a new column that shows it.

In [18]:
emergencies['year'] = emergencies['Week Start Date'].dt.year
emergencies

Week Start Date          Agency                      Final Incident Type  \
0         2014-12-29             EMS        1. Life Threating Med Emergencies   
1         2014-12-29             EMS  2. Non-Life Threatening Med Emergencies   
2         2014-12-29            FDNY                      1. Structural Fires   
3         2014-12-29            FDNY                  2. Non-Structural Fires   
4         2014-12-29            FDNY                   3. Medical Emergencies   
...              ...             ...                                      ...   
6343      2019-01-07  NYPD (Non-CIP)               Other Crimes (In Progress)   
6344      2019-01-07  NYPD (Non-CIP)                               Past Crime   
6345      2019-01-07  NYPD (Non-CIP)  Police Officer/Security Holding Suspect   
6346      2019-01-07  NYPD (Non-CIP)                          Possible Crimes   
6347      2019-01-07  NYPD (Non-CIP)                         Vehicle Accident   

      # of Incidents Calculated  Call to First Pickup  \
0                          8881                  3.02   
1                         12594                  3.03   
2                           341                  3.09   
3                           168                  3.07   
4                          3054                  3.02   
...                         ...                   ...   
6343                       2720                  3.58   
6344                       3200                  3.72   
6345                        195                  3.57   
6346                       7361                  3.56   
6347                       3179                  3.69   

      Call to PD Calltaker Handoff  Call to FDNY Pickup  \
0                            60.84                  NaN   
1                            69.63                  NaN   
2                            71.82                  NaN   
3                            80.86                  NaN   
4                           154.83                  NaN   
...                            ...                  ...   
6343                        152.76                  NaN   
6344                        205.11                  NaN   
6345                         99.61                  NaN   
6346                        148.94                  NaN   
6347                        173.26                  NaN   

      Call to FDNY Job Creation  Call to EMS Pickup  \
0                           NaN               68.10   
1                           NaN               77.76   
2                           NaN                 NaN   
3                           NaN                 NaN   
4                           NaN               66.45   
...                         ...                 ...   
6343                        NaN                 NaN   
6344                        NaN                 NaN   
6345                        NaN                 NaN   
6346                        NaN                 NaN   
6347                        NaN                 NaN   

      Call to Agency Job Creation  ...  \
0                          128.32  ...   
1                          125.53  ...   
2                             NaN  ...   
3                             NaN  ...   
4                          151.56  ...   
...                           ...  ...   
6343                          NaN  ...   
6344                          NaN  ...   
6345                          NaN  ...   
6346                          NaN  ...   
6347                          NaN  ...   

      Median Cumulative First Arrival  (Multi-Agency Incidents)  \
0                                                427.75           
1                                                571.50           
2                                                267.00           
3                                                327.00           
4                                                375.00           
...                                                 ...           
634

Now let’s filter our dataset to have data from 2016 to 2019 only

In [19]:
after2016= emergencies['Week Start Date'] >= '2016-01-01'

In [20]:
before2019= emergencies['Week Start Date'] < '2019-12-31'

In [21]:
emergencies_clean= emergencies[after2016 & before2019]
emergencies_clean

Week Start Date          Agency                      Final Incident Type  \
978       2016-12-26             EMS        1. Life Threating Med Emergencies   
979       2016-12-26             EMS  2. Non-Life Threatening Med Emergencies   
980       2016-12-26            FDNY                      1. Structural Fires   
981       2016-12-26            FDNY                  2. Non-Structural Fires   
982       2016-12-26            FDNY                   3. Medical Emergencies   
...              ...             ...                                      ...   
6343      2019-01-07  NYPD (Non-CIP)               Other Crimes (In Progress)   
6344      2019-01-07  NYPD (Non-CIP)                               Past Crime   
6345      2019-01-07  NYPD (Non-CIP)  Police Officer/Security Holding Suspect   
6346      2019-01-07  NYPD (Non-CIP)                          Possible Crimes   
6347      2019-01-07  NYPD (Non-CIP)                         Vehicle Accident   

      # of Incidents Calculated  Call to First Pickup  \
978                       10337                  4.02   
979                       13564                  3.99   
980                         362                  4.04   
981                         180                  3.61   
982                        2184                  3.58   
...                         ...                   ...   
6343                       2720                  3.58   
6344                       3200                  3.72   
6345                        195                  3.57   
6346                       7361                  3.56   
6347                       3179                  3.69   

      Call to PD Calltaker Handoff  Call to FDNY Pickup  \
978                          63.43                  NaN   
979                          78.65                  NaN   
980                          77.67                  NaN   
981                          82.58                  NaN   
982                         182.21                  NaN   
...                            ...                  ...   
6343                        152.76                  NaN   
6344                        205.11                  NaN   
6345                         99.61                  NaN   
6346                        148.94                  NaN   
6347                        173.26                  NaN   

      Call to FDNY Job Creation  Call to EMS Pickup  \
978                         NaN               68.19   
979                         NaN               84.74   
980                         NaN                 NaN   
981                         NaN                 NaN   
982                         NaN               70.48   
...                         ...                 ...   
6343                        NaN                 NaN   
6344                        NaN                 NaN   
6345                        NaN                 NaN   
6346                        NaN                 NaN   
6347                        NaN                 NaN   

      Call to Agency Job Creation  ...  \
978                        123.63  ...   
979                        128.43  ...   
980                           NaN  ...   
981                           NaN  ...   
982                        178.33  ...   
...                           ...  ...   
6343                          NaN  ...   
6344                          NaN  ...   
6345                          NaN  ...   
6346                          NaN  ...   
6347                          NaN  ...   

      Median Cumulative First Arrival  (Multi-Agency Incidents)  \
978                                              415.50           
979                                              532.25           
980                                              273.00           
981                                              318.50           
982                                              379.00           
...                                                 ...           
634

**1.2.6 Now that we fixed the time period, let’s go back to our goal: First, find out which incidents were the least common.**

Let’s create a dataframe that captures the count of records per Final Incident Type per Agency

In [22]:
counts= emergencies_clean.groupby(['Final Incident Type']).size().reset_index(name='count')
counts

Final Incident Type  count
0                                         1. Critical    209
1                   1. Life Threating Med Emergencies    209
2                                 1. Structural Fires    209
3             2. Non-Life Threatening Med Emergencies    209
4                             2. Non-Structural Fires    209
5                                          2. Serious    209
6                              3. Medical Emergencies    209
7                                     3. Non-Critical    209
8                          4. Non-Medical Emergencies    209
9                                              Alarms    209
10                      Disorderly Person/Group/Noise    209
11                                            Dispute    209
12  Hazardous Materials/Suspicious Letters/Package...    209
13                         Investigate/Possible Crime     22
14                         Other Crimes (In Progress)    209
15                                         Past Crime    209
16            Police Officer/Security Holding Suspect    209
17                                    Possible Crimes    209
18                                       Shot Spotter      4
19  United States Postal Service - Bio Hazard Dete...      6
20                                   Vehicle Accident    209

In [23]:
counts.sort_values('count', ascending=True) #Sorting it in ascending order

Final Incident Type  count
18                                       Shot Spotter      4
19  United States Postal Service - Bio Hazard Dete...      6
13                         Investigate/Possible Crime     22
0                                         1. Critical    209
17                                    Possible Crimes    209
16            Police Officer/Security Holding Suspect    209
15                                         Past Crime    209
14                         Other Crimes (In Progress)    209
12  Hazardous Materials/Suspicious Letters/Package...    209
11                                            Dispute    209
10                      Disorderly Person/Group/Noise    209
8                          4. Non-Medical Emergencies    209
7                                     3. Non-Critical    209
6                              3. Medical Emergencies    209
5                                          2. Serious    209
4                             2. Non-Structural Fires    209
3             2. Non-Life Threatening Med Emergencies    209
2                                 1. Structural Fires    209
1                   1. Life Threating Med Emergencies    209
9                                              Alarms    209
20                                   Vehicle Accident    209

The least common incidents were Shot Spotter and United States Postal Service Bio Hazard De- tection. However, the other incident types have the exact same count of incidents.

Let’s look at the original dataset to see if it was a probblem with data cleaning.

In [24]:
counts_emergencies= emergencies.groupby(['Final Incident Type']).size().reset_index(name='count')
counts_emergencies

Final Incident Type  count
0                                         1. Critical    349
1                   1. Life Threating Med Emergencies    351
2                                 1. Structural Fires    349
3             2. Non-Life Threatening Med Emergencies    351
4                             2. Non-Structural Fires    349
5                                          2. Serious    349
6                              3. Medical Emergencies    349
7                                     3. Non-Critical    349
8                          4. Non-Medical Emergencies    349
9                                              Alarms    349
10                      Disorderly Person/Group/Noise    349
11                                            Dispute    349
12  Hazardous Materials/Suspicious Letters/Package...    349
13                         Investigate/Possible Crime     47
14                         Other Crimes (In Progress)    349
15                                         Past Crime    349
16            Police Officer/Security Holding Suspect    349
17                                    Possible Crimes    349
18                                       Shot Spotter      7
19  United States Postal Service - Bio Hazard Dete...      8
20                                   Vehicle Accident    349

In [25]:
counts_emergencies.sort_values('count', ascending=True)

Final Incident Type  count
18                                       Shot Spotter      7
19  United States Postal Service - Bio Hazard Dete...      8
13                         Investigate/Possible Crime     47
0                                         1. Critical    349
17                                    Possible Crimes    349
16            Police Officer/Security Holding Suspect    349
15                                         Past Crime    349
14                         Other Crimes (In Progress)    349
12  Hazardous Materials/Suspicious Letters/Package...    349
11                                            Dispute    349
10                      Disorderly Person/Group/Noise    349
8                          4. Non-Medical Emergencies    349
7                                     3. Non-Critical    349
6                              3. Medical Emergencies    349
5                                          2. Serious    349
4                             2. Non-Structural Fires    349
2                                 1. Structural Fires    349
9                                              Alarms    349
20                                   Vehicle Accident    349
3             2. Non-Life Threatening Med Emergencies    351
1                   1. Life Threating Med Emergencies    351

Most of the counts are 349, but we still see that Shot Spotter and USPS Bio Hazard as the top two incident types with the lowest count. What agency is associacted with these two incident types? Let’s find out.

In [26]:
agencies= emergencies_clean.groupby('Agency').size().reset_index(name='count')
agencies

Agency  count
0             EMS    418
1            FDNY    836
2            NYPD    627
3  NYPD (Non-CIP)   1913

**1.2.7 Using Boolean Indexing**

*EMS*


How it works: Create a variable called ems. Locate the series labeled ‘Agency’ in the dataframe. Once ‘Agency’ is found, test if the agency is EMS. Create a new dataframe that shows the previous two conditions

In [27]:
ems=emergencies_clean[emergencies_clean['Agency']=='EMS']

In [28]:
 count_ems=ems.groupby('Final Incident Type').size().reset_index(name='count').sort_values('Final Incident Type')
count_ems

Final Incident Type  count
0        1. Life Threating Med Emergencies    209
1  2. Non-Life Threatening Med Emergencies    209

Repeat the same procedure but now with the other agencies

*FDNY*

In [29]:
fdny=emergencies_clean[emergencies_clean['Agency']=='FDNY']

In [30]:
 count_fdny=fdny.groupby('Final Incident Type').size().reset_index(name='count').sort_values('Final Incident Type')
count_fdny

Final Incident Type  count
0         1. Structural Fires    209
1     2. Non-Structural Fires    209
2      3. Medical Emergencies    209
3  4. Non-Medical Emergencies    209

*NYPD*

In [31]:
nypd=emergencies_clean[emergencies_clean['Agency']=='NYPD']
 

In [32]:
count_nypd=nypd.groupby('Final Incident Type').size().reset_index(name='count').sort_values('Final Incident Type')
count_nypd

Final Incident Type  count
0         1. Critical    209
1          2. Serious    209
2     3. Non-Critical    209

*NYPD (Non-CIP)*

In [33]:
nypd_non_CIP=emergencies_clean[emergencies_clean['Agency']=='NYPD (Non-CIP)']

In [34]:
 count_nypd_non_CIP=nypd_non_CIP.groupby('Final Incident Type').size().reset_index(name='count').sort_values('Final Incident Type')
count_nypd_non_CIP

Final Incident Type  count
0                                              Alarms    209
1                       Disorderly Person/Group/Noise    209
2                                             Dispute    209
3   Hazardous Materials/Suspicious Letters/Package...    209
4                          Investigate/Possible Crime     22
5                          Other Crimes (In Progress)    209
6                                          Past Crime    209
7             Police Officer/Security Holding Suspect    209
8                                     Possible Crimes    209
9                                        Shot Spotter      4
10  United States Postal Service - Bio Hazard Dete...      6
11                                   Vehicle Accident    209

In [35]:
count_nypd_non_CIP.sort_values('count', ascending=True) #Sorting

Final Incident Type  count
9                                        Shot Spotter      4
10  United States Postal Service - Bio Hazard Dete...      6
4                          Investigate/Possible Crime     22
0                                              Alarms    209
1                       Disorderly Person/Group/Noise    209
2                                             Dispute    209
3   Hazardous Materials/Suspicious Letters/Package...    209
5                          Other Crimes (In Progress)    209
6                                          Past Crime    209
7             Police Officer/Security Holding Suspect    209
8                                     Possible Crimes    209
11                                   Vehicle Accident    209

Results show that the NYPD (Non-CIP) is the agency with the lowest incident type count.

**1.2.8 Now, let’s dive into the second part of today’s goal:**


Discover which agencies responded to the least number of incidents from 2016 to 2019.

First, upload necessary visualization packages


In [36]:
pip install -U kaleido

Note: you may need to restart the kernel to use updated packages.


In [37]:
 import plotly.io as pio
pio.renderers.default = "notebook_connected+pdf"

In [38]:
import plotly.express as px

Now, let's plot the # of Incident Calculated over time

In [39]:
fig = px.box(emergencies_clean, x="year", y="# of Incidents Calculated",color="Agency")
fig.show()

The minimum number of incidents from 2016 2019 correponds to NYPD (Non-CIP).